Competition/src/main/cpp/subsystems/Shooter.cpp

```diff
@@ -219,51 +240,42 @@ void Shooter::analyzeDashboard()
 }
 
 void Shooter::assignOutputs()
-{   
-    state.turretOutput = 0;
-    state.turretTarget = 0;
-
-    state.flywheelTarget = 0;
-    state.hoodTarget = 0;
+{ 
 
-    bool useSmartMotion = false;
+    /*//////////////////////////////////////
+    // Turret                             //  
+    //////////////////////////////////////*/
 
     //MANUAL
     if (state.turretState == TurretState::TURRET_MANUAL) {
-        int sign = 1;//state.leftStickX >= 0 ? 1 : -1;
-        state.turretOutput = sign * std::pow(state.leftStickX, 3) * ShooterConstants::TURRET_SPEED_MULTIPLIER;
+        state.turretOutput = std::pow(state.leftStickX, 3) * ShooterConstants::TURRET_SPEED_MULTIPLIER;
 
         // Minimum power deadband
         if (std::abs(state.leftStickX) < ShooterConstants::pDeadband) {
             state.turretOutput = 0;
         }
         // Stop deadband
-        else if (std::abs(state.turretOutput) < ShooterConstants::pSoftDeadband) {
-            int direction = 1;
-            if (state.turretOutput < 0) direction = -1;
-            state.turretOutput = ShooterConstants::pSoftDeadband * direction;
-        }
+        // else if (std::abs(state.turretOutput) < ShooterConstants::pSoftDeadband) {
+        //     int direction = 1;
+        //     if (state.turretOutput < 0) direction = -1;
+        //     state.turretOutput = ShooterConstants::pSoftDeadband * direction;
+        // }
+        turret.Set(state.turretOutput);
     }
-
     //HOME
     else if(state.turretState == TurretState::TURRET_HOME){
         state.turretTarget = ShooterConstants::homePosition - turretEncoder.GetPosition();
-        useSmartMotion = false;
+        turret.Set(state.turretOutput);
```

Comment:
I think this should use smart motion and target the calculated turretTarget encoder position. Right now it is just setting an unrelated percentage output to the motor (state.turretOutput).

Reply:
I completely agree with you, except for now we have only tested the custom PID stuff with the turret.

So the idea would be to use this for now, and then you guys can replace this with the smart motion logic later this week

Reply:
Even if we aren't to use the smart motion, shouldn't we be using the duty cycle control? We still want to have that PID control. Additionally, here we're just setting it to turretOutput, which isn't being updated. We're only calculating the target.